src/pybind/mgr/dashboard/controllers/nfsganesha.py

```diff
@@ -131,29 +131,31 @@ def list(self):
     def create(self, path, cluster_id, daemons, pseudo, tag, access_type,
                squash, security_label, protocols, transports, fsal, clients,
                reload_daemons=True):
-        if fsal['name'] not in Ganesha.fsals_available():
+        if fsal['name'] not in mgr.remote('nfs', 'cluster_fsals'):
             raise NFSException("Cannot create this export. "
                                "FSAL '{}' cannot be managed by the dashboard."
                                .format(fsal['name']))
 
-        ganesha_conf = GaneshaConf.instance(cluster_id)
-        ex_id = ganesha_conf.create_export({
+        fsal.pop('user_id')  # mgr/nfs does not let you customize user_id
+        # FIXME: what was this?     'tag': tag,
+        raw_ex = {
             'path': path,
             'pseudo': pseudo,
             'cluster_id': cluster_id,
             'daemons': daemons,
-            'tag': tag,
             'access_type': access_type,
             'squash': squash,
             'security_label': security_label,
             'protocols': protocols,
             'transports': transports,
             'fsal': fsal,
             'clients': clients
-        })
-        if reload_daemons:
-            ganesha_conf.reload_daemons(daemons)
-        return ganesha_conf.get_export(ex_id).to_dict()
+        }
+        export_mgr = mgr.remote('nfs', 'fetch_nfs_export_obj')
+        ret, out, err = export_mgr.apply_export(cluster_id, json.dumps(raw_ex))
```

Comment:
Why not exposing this as a single method?
```suggestion
        ret, out, err = mgr.remote('nfs', 'apply_export', cluster_id, raw_ex)
```

Reply:
Because `apply_export()` is a method of `ExportMgr`.

Reply:
https://github.com/ceph/ceph/blob/f435859ec4ee4b5dc9903c209333ddbd17f7e1da/src/pybind/mgr/nfs/module.py#L28-L42